### PR TITLE
Further receive tests development

### PR DESCRIPTION
**Pull request by cmrd-senya**
With this PR I'll post updates to the receive integration tests until they become comprehensive.

### CODE REVIEW

File: spec/integration/federation/federation_messages_generation.rb [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  times = 1
+  if %i(signed_retraction_entity relayable_retraction_entity).include?(entity_name)
+    times += 2
+  end
+
+  substitute_wrong_key(sender, times)
+  generate_retraction(entity_name, target_object, sender)
+end
+
+def generate_relayable_local_parent(entity_name)
```
Comment (diaspora-code-review) [inline spec/integration/federation/federation_messages_generation.rb]: Assignment Branch Condition size for generate_relayable_local_parent is too high. [22.11/20]

File: spec/integration/federation/federation_messages_generation.rb [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                                            .with(
+                                              :fetch_author_private_key_by_entity_guid,
+                                              klass.get_target_entity_type(@entity.to_h),
+                                              kind_of(String)
+                                            )
+                                            .and_return(nil)
+  generate_xml(@entity, @remote_user, @user)
+end
+
+def generate_relayable_remote_parent(entity_name)
```
Comment (diaspora-code-review) [inline spec/integration/federation/federation_messages_generation.rb]: Assignment Branch Condition size for generate_relayable_remote_parent is too high. [24.1/20]

**Comment by cmrd-senya** [comment]
Ok, I've brought back the glue code to the place it is supposed to be.

**Comment by SuperTux88** [comment]
I rebased your branch without the revert-commit: https://github.com/SuperTux88/diaspora/tree/receive-tests (maybe you can `git reset` this branch to this rebase? then we have no back and forward stuff). I try to redo my refactorings (but you have even more global variables now :( it's really complicated to find where there are defined) on top of this rebased branch as my own commit and create a PR to your PR-branch then.

**Comment by cmrd-senya** [comment]
I don't see something bad in having a revert commit, but if you would
like I'll rebase it onto your branch.

I don't really like the fact you moved methods from
federation_messages_generation.rb to different other places. I think the
structure is pretty clear now. There are 3 files for shared behaviour,
federation_messages_generation for xml test messages producing functions
and the main test file. I don't like the idea of ungrouping functions
from federation_messages_generation. Maybe put them into a Module, so
navigation is easier?

**Comment by cmrd-senya** [comment]
Ok, I think this is finished for me.
We can merge it, unless @SuperTux88 doesn't want to add any refactoings on top.

**Comment by SuperTux88** [comment]
I finished the refactoring on my branch: https://github.com/SuperTux88/diaspora/tree/receive-tests

Can you please `git reset` your branch to my branch (or I can create my own PR)

The refactoring contains:
* less global methods (the code is where it is needed)
* no global `@entity` (all needed variables as parameters), this helps to find where something comes from
* 129 fewer lines of code
* maybe some small code duplications for better readability

**Comment by cmrd-senya** [comment]
Done, thanks

**Comment by SuperTux88** [comment]
thanks

**Comment by denschub** [comment]
Merged, thanks.

**Comment by cmrd-senya** [comment]
Cool, thanks!

**Pull request closed without merging**